### PR TITLE
Update use cases of Developer Guide v1.1

### LINKED ISSUE #4: Update use cases
Update use cases to reflect the user journey map for AIA

### PR DESCRIPTION
**Pull request by ngjunkang**
This commit will close #4 by updating use cases to reflect the user
journey map for AIA.

Let's add the following use cases,
* Add a new contact
* Update a contact
* List all contacts in Address Book
* Delete a contact
* Find contact by keyword
* Set previously contacted date
* List all contacts in Address Book within a designated time frame
* Set reminder
* View reminders of a contact
* View reminders on a date
* Delete a reminder
* Show all tags
* Show all contacts that are tagged to a tag
* Add a tag to a contact
* Delete a tag from a contact

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -283,31 +283,297 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+      Use case ends.
+
+* 1b. The given index is invalid.
+
+    * 1b1. Address Book shows an error message.
+
+      Use case ends.
+
+**Use case: UC03 - List all contacts in Address Book**
```
Comment (bloodofme) [inline docs/DeveloperGuide.md:328]: Better to add an extension for the case whereby user tries to list an empty address book

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -283,31 +283,297 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1. User requests to list all contacts (UC03).
+2. Address Book shows a list of contacts (UC03).
+3. User requests to delete a specific contact in the list.
+4. Address Book deletes the contact.
+
+   Use case ends.
 
 **Extensions**
 
 * 2a. The list is empty.
```
Comment (bloodofme) [inline docs/DeveloperGuide.md:350]: Would AIA show an error message ?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -283,31 +283,297 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
     * 3a1. AddressBook shows an error message.
 
       Use case resumes at step 2.
 
+**Use case: UC05 - Find contact by keyword**
+
+**MSS**
+
+1. User requests to find a person.
+2. Address Book shows a list of persons with names containing the keyword, if any.
```
Comment (bloodofme) [inline docs/DeveloperGuide.md:365]: Add an extension for the case where keyword does not match any contact ?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -283,31 +283,297 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+    * 1c1. Address Book shows an error message.
+
+      Use case ends.
+
+**Use case: UC07 - List all contacts in Address Book within a designated time frame**
+
+**MSS**
+
+1. User requests to see all contacts last contacted within a designated number of days.
```
Comment (bloodofme) [inline docs/DeveloperGuide.md:410]: Add an extension for a case whereby there are no contacted clients within the designated number of days

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -283,31 +283,297 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+    * 1a1. Address Book shows an error message.
+
+      Use case ends.
+
+**Use case: UC09 - View reminders of a contact**
+
+**MSS**
+
+1. User requests to view reminders of a specific contact.
```
Comment (bloodofme) [inline docs/DeveloperGuide.md:444]: Add extension for case whereby theres no reminder for a specific contact

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -283,31 +283,297 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Extensions**
+
+* 1a. The given input is invalid.
+
+    * 1a1. Address Book shows an error message.
+
+      Use case ends.
+
+**Use case: UC13 - Show all contacts that are tagged to a tag**
```
Comment (bloodofme) [inline docs/DeveloperGuide.md:516]: Add extension for invalid tag

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -283,31 +283,297 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+      Use case ends.
+
+* 1c. The given tag already exists.
+
+    * 1c1 Address Book shows an error message.
+
+      Use case ends.
+
+**Use case: UC15 - Delete a tag from a contact**
```
Comment (bloodofme) [inline docs/DeveloperGuide.md:554]: Add extension for invalid tag

**Review by bloodofme — COMMENTED** [review]
Very detailed. Very nice.

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -283,31 +283,297 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**Use case: UC08 - Set reminder**
+
+**MSS**
+
+1. User requests to set a reminder for a designated contact at a specified date.
+2. Address Book sets a reminder for the contact tagged to the specified date.
+
+   Use case ends.
+
+**Extensions**
```
Comment (cashewnade) [inline docs/DeveloperGuide.md:432]: Perhaps can add an extension for a given date is invalid?

**Pull request merged**